scipy/sparse/linalg/tests/test_interface.py

```diff
@@ -285,11 +288,20 @@ def check_matvec(
         """
         rng = np.random.default_rng(42)
 
-        u = rng.standard_normal(op.shape[-1], dtype=data_dtype)
-        v = rng.standard_normal(op.shape[-2], dtype=data_dtype)
+        dtype = np.dtype(data_dtype)
+        u = rng.standard_normal(op.shape[-1], dtype=dtype)
+        v = rng.standard_normal(op.shape[-2], dtype=dtype)
         if complex_data:
-            u = u + (1j * rng.standard_normal(op.shape[-1], dtype=data_dtype))
-            v = v + (1j * rng.standard_normal(op.shape[-2], dtype=data_dtype))
+            u = u + (1j * rng.standard_normal(op.shape[-1], dtype=dtype))
+            v = v + (1j * rng.standard_normal(op.shape[-2], dtype=dtype))
+        
+        # TODO: handle empty batches
+        # TODO: test vectors with different but broadcastable batch shapes?
+        # Test `u` and `v` with the same batch shape as `op`
+        if batch_shape := op.shape[:-2]:
+            batch_scale = rng.standard_normal((*batch_shape, 1), dtype=dtype)
+            u = batch_scale * u
+            v = batch_scale * v
```

Comment:
This works even if the batch is empty, right? I got to thinking about it because I still don't read the walrus fluently and had to think.
```suggestion
        batch_shape = op.shape[:-2]
        batch_scale = rng.standard_normal((*batch_shape, 1), dtype=dtype)
        u = batch_scale * u
        v = batch_scale * v
```
Would the test not pass if `u` and `v` were arbitrary - the batches need to be formed by scaling the original `u` and `v`? Same as below.